Dockerfile

```diff
@@ -16,12 +16,36 @@ RUN sudo apt-get install -y \
   lsof \
   jq \
   groff \
-  python \
-  python-pip \
-  libpython-dev \
   less \
-  tree
+  tree \
+  nano
 
+# Install NodeJS tools
+RUN mkdir /home/circleci/.npm-global && \
+    mkdir /home/circleci/.npm-global/lib && \
+    sudo npm install -g create-react-app
```

Comment:
That doesn't seem to be required.
I added these lines because NPM was looking for these dirs and was failing in couple of e2e tests that were spinning up react apps.

Reply:
If that works, then sounds good to me. It's possible CircleCI is setting this somehow, though I don't see anything in their [`Dockerfile` that indicates that](https://github.com/CircleCI-Public/cimg-node/blob/main/18.14/Dockerfile).

Reply:
This has been found experimentally. Couple of tests that spin up react app were calling into npm/npx and it complained about missing `/home/circleci/.npm-global/lib` directory, creating it mitigated it.
I don't know why base image starts with broken cache.
the installation of create-react-app is unrelated optimization.